### ISSUE TITLE
bug:  okta source plugin unable to unmarshal okta_applications tables

### ISSUE THREAD

**Issue by nkvon** [cloudquery/cloudquery#16836]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

okta_application tables are not getting synced. I am seeing the following errors.

### Expected Behavior

it should sync data into okta_application_users & okta_application_group_assignments tables

### CloudQuery (redacted) config

```spec:
  # Source spec section
  name: okta
  path: cloudquery/okta
  registry: cloudquery
  version: "v4.0.1"
  tables: ["*"]
  destinations: ["postgresql"]
  spec:
    # Required. Your Okta domain name
    domain: <<Domain Name>>

    # Optional. Okta Token to access API
    # ⚠️ Warning - Your token should be kept secret and not committed to source control.
    # ⚠️ Warning - In the future versions token parameter will become required.
    token: << Token >>

    # Optional. Rate limiter settings
    concurrency: 200
    rate_limit:
      max_backoff: 65s
      max_retries: 3```

### Steps To Reproduce

use this configuration and run sync.

### CloudQuery (redacted) logs

```truncated plugin log line line="{"level":"error","module":"okta-src","table":"okta_applications","client":"okta","error":"received error from Okta API: err: Failed to unmarshal ListApplications200ResponseInner as OpenIdConnectApplication: json: cannot unmarshal bool into Go struct field _Application.profile of type map[string]interface {}, body:```

### CloudQuery version

3.29.1

### Additional Context

_No response_

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by erezrokah** [comment]
Worth updating to the latest API https://github.com/cloudquery/cloudquery/pull/16831

**Comment by disq** [comment]
@nkvon Hi, thanks for reporting. I'm having trouble reproducing this, can you please tell me which Okta application (or is it a custom one?) are you having this issue with?

In the meantime we've released [Okta source 4.1.0](https://hub.cloudquery.io/plugins/source/cloudquery/okta) which updates the Okta SDK, can you give it a try?

**Comment by nkvon** [comment]
@disq Hi,, Sure i will give it a try. i am trying to data at okta org level. so it is for all and any applications configured in the okta org. I will post my results after trying 4.1.0. thanks a lot.

**Comment by disq** [comment]
@nkvon Looks like it's having trouble with an `OpenIdConnectApplication`... To reproduce, I tried setting up an OIDC IdP as test but didn't even get an entry in applications. So my question is, which apps you have installed might fall into this category. (Let me know if 4.1.0 helps as well)

**Comment by erezrokah** [comment]
Closing as stale, @nkvon if you're still hitting this issue please comment and I'll re-open

**Comment by komodo472** [comment]
@erezrokah I think I'm running into the same issue

cloudquery version 5.11.0

Config
```
kind: source
spec:
  name: okta
  path: cloudquery/okta
  registry: cloudquery
  version: "v4.1.6"
  tables: 
    - okta_applications
  destinations: ["postgresql"]
  spec:
    domain: "https://test.okta.com"
    token: "${OKTA_TOKEN}"

```

Error
```
ERR table resolver finished with error error="received error from Okta API: err: json: cannot unmarshal string into Go struct field _ApplicationGroupAssignment.profile of type map[string]interface {}, body: \"[{\\\"id\\\":\\\"00gos0vfw6WxG2yQu350\\\",\\\"lastUpdated\\\":\\\"2024-01-31T19:11:22.000Z\\\",\\\"priority\\\":0,\\\"profile\\\":{\\\"role\\\":\\\"CEO\\\",\\\"salesforceGroups\\\":null,\\\"city\\\":null,\\\"profile\\\":\\\"System Administrator\\\",\\\"postalCode\\\":null,\\\"companyName\\\":null,\\\"featureLicenses\\\":null,\\\"division\\\":null,\\\"street\\\":null,\\\"state\\\":null,\\\"department\\\":null,\\\"publicGroups\\\":null},\\\"_links\\\":{\\\"app\\\":{\\\"href\\\":\\\"https://test.okta.com/api/v1/apps/0oa108e9xpmjo7eId350\\\"},\\\"self\\\":{\\\"href\\\":\\\"https://test.okta.com/api/v1/apps/0oa108e9xpmjo7eId350/groups/00gos0vfw6WxG2yQu350\\\"},\\\"group\\\":{\\\"href\\\":\\\"https://test.okta.com/api/v1/groups/00gos0vfw6WxG2yQu350\\\"}}}]\" model: <nil>" client=okta invocation-id=5b38b1f4-9cc3-4991-b231-a7454d2f85c3 module=okta-source table=okta_application_group_assignments

Parsed profile from error log:
[
    {
        "id": "00gos0vfw6WxG2yQu350",
        "lastUpdated": "2024-01-31T19:11:22.000Z",
        "priority": 0,
        "profile": {
            "role": "CEO",
            "salesforceGroups": null,
            "city": null,
            "profile": "System Administrator",
            "postalCode": null,
            "companyName": null,
            "featureLicenses": null,
            "division": null,
            "street": null,
            "state": null,
            "department": null,
            "publicGroups": null
        },
        "_links": {
            "app": {
                "href": "https://test.okta.com/api/v1/apps/0oa108e9xpmjo7eId350"
            },
            "self": {
                "href": "https://test.okta.com/api/v1/apps/0oa108e9xpmjo7eId350/groups/00gos0vfw6WxG2yQu350"
            },
            "group": {
                "href": "https://test.okta.com/api/v1/groups/00gos0vfw6WxG2yQu350"
            }
        }
    }
]
```

It seems like an issue with custom fields in the profile. Every app I sync gets an error for either `_AppUser.profile` or `_ApplicationGroupAssignment.profile`

**Comment by erezrokah** [comment]
A fixed for the issue has been released in [`v4.1.7`](https://github.com/cloudquery/cloudquery/releases/tag/plugins-source-okta-v4.1.7)
